tests/test_endpoint_buffers.py

```diff
@@ -139,10 +140,9 @@ def test_prepare_results_for_publish(forward_setup):
             }
         }
 
-    @staticmethod
     @patch("gsy_e.gsy_e_core.sim_results.endpoint_buffer.get_json_dict_memory_allocation_size")
     def test_prepare_results_for_publish_output_too_big(
-            get_json_dict_memory_allocation_size_mock, forward_setup, caplog):
+            self, get_json_dict_memory_allocation_size_mock, forward_setup, caplog):
```

Comment:
Why is this change needed?

Reply:
That's something that gave me some headaches. I wanted to create a class (`TestCoefficientEndpointBuffer`) that inherited from `TestSimulationEndpointBuffer` in order to automatically run all the tests that are declared in the parent class.
Unfortunately, if you try to do that when the parent class has static methods, this will break up stuff. For some reason (I suppose something internal to pytest) inherited methods that are declared as `@staticmethod` will not work (they raise an error). That seems to be related to how pytest works. I'm not sure whether this has been fixed in recent releases.

Reply:
Interesting. Thanks!